relationalize/sql_dialects.py

```diff
@@ -1,41 +1,57 @@
-from typing import Dict, List
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from typing import Generic, Literal, NewType, TypeVar, override
 
-_COLUMN_SEPERATOR = "\n    , "
+from relationalize.types import SupportedColumnType
 
+_COLUMN_SEPARATOR = "\n    , "
 
-class SQLDialect:
+DDLColumn = NewType('DDLColumn', str)
+DialectColumnType = TypeVar('DialectColumnType')
+
+class SQLDialect(ABC, Generic[DialectColumnType]):
     """
     Parent class for different sql dialects.
 
     Child classes must implement the `generate_ddl_column` method
     , and provide `type_column_mapping` and `base_ddl`.
     """
 
-    type_column_mapping: Dict[str, str]
+    type_column_mapping: Mapping[SupportedColumnType, DialectColumnType]
     base_ddl: str
 
     @staticmethod
-    def generate_ddl_column(column_name: str, column_type: str):
+    @abstractmethod
+    def generate_ddl_column(column_name: str, column_type: DialectColumnType) -> DDLColumn:
         raise NotImplementedError()
 
-    def generate_ddl(self, schema: str, table_name: str, columns: List[str]):
+    def generate_ddl(self, schema: str, table_name: str, columns: list[str]):
         """
         Generates a complete "Create Table" statement given the
         schema, table_name, and column definitions.
         """
-        columns_str = _COLUMN_SEPERATOR.join(columns)
+        columns_str = _COLUMN_SEPARATOR.join(columns)
         return self.base_ddl.format(
             schema=schema, table_name=table_name, columns=columns_str
         )
 
 
-class PostgresDialect(SQLDialect):
+PostgresColumn = Literal[
+    'BIGINT',
+    'BOOLEAN',
+    'FLOAT',
+    'TIMESTAMP',
+    'VARCHAR(65535)',
+]
+
+class PostgresDialect(SQLDialect[PostgresColumn]):
     """
     Inherits from `SQLDialect` and implements the postgres syntax.
     """
 
-    type_column_mapping: Dict[str, str] = {
+    type_column_mapping: Mapping[SupportedColumnType, PostgresColumn] = {
         "int": "BIGINT",
+        "datetime": "TIMESTAMP",
```

Comment:
this is the only place where actual business logic was altered

Reply:
I like this - I like the type hints, I like the additional timestamp support. But, I'm unsure what it will do to every existing pipeline that was already affected by this. let me explore a bit

Reply:
Alright you've got our go-ahead on this. Low chance that there's some edge case this breaks we haven't considered, and even lower chance that we need to revert from 0.1.8 back to 0.1.7